### PR TITLE
Fix the AUTO loadnetwork failure triggered  by AUTO:-xxx

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Check if the device is supported when AUTO retrieves the device list based on the ov::device::priorities.

### Tickets:
 - 99669

### CODE REVIEW

File: src/tests/unit/auto/get_device_list.cpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ const char igpuFullDeviceName[] = "Intel(R) Gen9 HD Graphics (iGPU)";
 const char dgpuFullDeviceName[] = "Intel(R) Iris(R) Xe MAX Graphics (dGPU)";
 // const char myriadFullDeviceName[] = "Intel Movidius Myriad X VPU";
 // const char vpuxFullDeviceName[] = "";
-const std::vector<std::string>  availableDevs = {"CPU", "GPU.0", "GPU.1", "VPUX"};
+const std::vector<std::string> availableDevs = {"CPU", "GPU.0", "GPU.1", "VPUX", "UNSUPPORTED_DEVICE"};
```
Comment (wangleis) [inline src/tests/unit/auto/get_device_list.cpp:36]: I guess "UNSUPPORTED_DEVICE" needs different behavior compared to other available devices

**Comment by peterchen-intel** [comment]
Add NVIDIA back and remove MYRIAD

**Review by peterchen-intel — APPROVED** [review]

File: src/plugins/auto/plugin.cpp [hunk 2]
```diff
@@ -887,12 +884,17 @@ std::string MultiDeviceInferencePlugin::GetDeviceList(const std::map<std::string
                 updateDeviceVec(iter);
             }
             for (auto&& device : deviceVec) {
-                allDevices += device;
-                allDevices += ((device == deviceVec[deviceVec.size()-1]) ? "" : ",");
+                if (!_pluginConfig.isSupportedDevice(device))
+                    continue;
```
Comment (songbell) [inline src/plugins/auto/plugin.cpp:888]: I'm thinking, how can you judge if device is supported when in future we remove the white list(which was originally used for secondary property checking?)

Comment (peterchen-intel) [reply src/plugins/auto/plugin.cpp:888]: Will check if device is in blacklist in future, currently is checking if device is in whitelist.

**Review by wangleis — APPROVED** [review]

**Pull request merged**